### DISCUSSION TITLE
Detailed mesmerizer pint out/ beginner starting points

### DISCUSSION THREAD

**Discussion by oni8245** [PlummersSoftwareLLC/NightDriverStrip#470] [Q&A]
Is there a document that has info on the mesmerizer board itself? what the buttons do and pins do etc...
Also a bit new to this I've got it working on the led matrix. But where do i start to upload my own to the webserver? 
Is it done over the serial port? I'm sure theses have been asked before but i wisent able to find anything out there. 
Thank you all!

**Comment by robertlipe** [comment]
On Tue, Oct 24, 2023, 12:31 PM oni8245 ***@***.***> wrote:

> Is there a document that has info on the mesmerizer board itself? what the
> buttons do and pins do etc...
>

The buttons are the gpio/bootstrapping pins that are on most every esp32
board. Unless you have a module that's fresh from manufacturer, you can
safely ignore them.  EN doubles as a reset.

I've tried to get documentation on the headers and haven't been
successful.  The silk screen on the big header is generally clear enough
that you can add hardware jtag or snag a gpio if you want to add some
custom sensor.

From what I can tell, the Mesmerizer hardware really is meant to be more of
a turnkey product than an open hacking platform.  The integration of mic
and audio in and such seem more the focus than offering expansion pins.

Also a bit new to this I've got it working on the led matrix. But where do
> i start to upload my own to the webserver?
> Is it done over the serial port? I'm sure theses have been asked before
> but i wisent able to find anything out there.
>

You can upload code over the USB/serial port or over the network itself,
but Mesmerizer is too slow for that to be practical for development.  Other
esp32 modules are way faster for network code upload.

There's a bunch of doc within a few clicks at
https://github.com/PlummersSoftwareLLC/NightDriverStrip/tree/5a78b33ae34c59d36f085b71336acc77cc5e714a

>
